src/task_list/time_integrator.cpp

```diff
@@ -966,6 +967,7 @@ TimeIntegratorTaskList::TimeIntegratorTaskList(ParameterInput *pin, Mesh *pm) {
       } else {
         AddTask(INT_SCLR,CALC_SCLRFLX);
       }
+      AddTask(INT_CHM,INT_SCLR);
```

Comment:
This is not for @munan but rather my personal note, but I feel the ordering of the task list construction can be improved. Currently some tasks appear in dependencies before they are added to the list. It works as long as the dependencies are correctly set, but not necessarily optimal (and can be confusing).